Source/DataModels/Resources/ResourceTexture.cpp

```diff
@@ -60,17 +60,17 @@ void ResourceTexture::CreateTexture()
 {
 	glGenTextures(1, &glTexture);
 	glBindTexture(GL_TEXTURE_2D, glTexture);
+
+	glTexImage2D(GL_TEXTURE_2D, 0, internalFormat, width, height, 0, format, imageType, &(pixels[0]));
+
+	if (loadOptions.mipMap) glGenerateMipmap(GL_TEXTURE_2D);
```

Comment:
Better to put it in a separate line.

```suggestion
	if (loadOptions.mipMap)
	{
		glGenerateMipmap(GL_TEXTURE_2D);
	}
```